test/e2e/framework/scaletest/get-publish-metrics.go

```diff
@@ -43,12 +56,36 @@ func (g *GetAndPublishMetrics) Run() error {
 		g.telemetryClient = telemetryClient
 	}
 
+	config, err := clientcmd.BuildConfigFromFlags("", g.KubeConfigFilePath)
+	if err != nil {
+		return fmt.Errorf("error building kubeconfig: %w", err)
+	}
+
+	k8sClient, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		return fmt.Errorf("error creating Kubernetes client: %w", err)
+	}
+	g.k8sClient = k8sClient
+
+	metricsClient, err := metrics.NewForConfig(config)
+	if err != nil {
+		return fmt.Errorf("error creating metrics client: %w", err)
+	}
+	g.metricsClient = metricsClient
+
 	g.stop = make(chan struct{})
 	g.wg.Add(1)
 
 	go func() {
 
-		t := time.NewTicker(5 * time.Minute)
+		t := time.NewTicker(defaultInterval)
+
+		// First execution
+		err := g.getAndPublishMetrics()
+		if err != nil {
+			log.Fatalf("error getting and publishing number of restarts: %v", err)
+			return
```

Comment:
There's no point in returning since `log.Fatalf` will panic. That said, you should not panic :)